lib/PrintAsClang/DeclAndTypePrinter.cpp

```diff
@@ -1440,6 +1440,7 @@ class DeclAndTypePrinter::Implementation
     os << "SWIFT_EXTERN ";
     printFunctionDeclAsCFunctionDecl(FD, FD->getCDeclName(), resultTy);
     printFunctionClangAttributes(FD, funcTy);
+    os << " SWIFT_NOEXCEPT";
```

Comment:
Shouldn't `noexcept` be the first thing after the function declaration and before any possible attributes? Does the test below try to parse the header or just checks the output?

Reply:
Yeah, the `RUN: %check-interop-cxx-header-in-clang(%t/functions.h)` runline will try to parse the header. I used this order for consistency, because other parts of the codebase was also emitting those attributes first and noexcept later. I think there is no strict place to put these non-standard attributes on the declaration, Clang seems to be forgiving on that front.

Reply:
The parsing is not as forgiving on my side. In Xcode 16b2 if I add

```
extern int _Nonnull foo(void) __attribute__((warn_unused_result)) noexcept;
```

to a C++ file, I get:

```
error: expected function body after function declarator
   10 | extern int _Nonnull foo(void) __attribute__((warn_unused_result)) noexcept;
      |                                                                   ^
```

Where does the `noexcept` is added in other parts of the codebase? I just sent #74780, but if there's more places it might not be bad to look at them too. They might need fixing.

Reply:
I just checked and it looks like the order was an oversight on my part. Too bad the `-fsyntax-only` wasn't good enough to catch this :/ Thanks a lot for the PR, I just approved it. We also need to cherry-pick this to the release branch.